asv_bench/benchmarks/benchmarks.py

```diff
@@ -819,4 +821,147 @@ def time_index(self, shape):
+        self.data.loc[800]
+
+    def time_loc_slice(self, shape, index, index_structure):
+        execute(self.data.loc[:800])
+
+
+class TimeReindex:
+    param_names = ["shape"]
+    params = [get_benchmark_shapes("TimeReindex")]
+
+    def setup(self, shape):
+        Rows, Cols = shape
+        rng = IMPL.date_range(start="1/1/1970", periods=Rows, freq="1min")
+        self.df = DataFrame(np.random.rand(Rows, Cols), index=rng, columns=range(Cols))
+        self.df["foo"] = "bar"
+        self.rng_subset = Index(rng[::2])
+        self.df2 = DataFrame(
+            index=range(Rows), data=np.random.rand(Rows, Cols), columns=range(Cols)
+        )
+        N = Rows
+        K = Cols
+        level1 = tm.makeStringIndex(N).values.repeat(K)
+        level2 = np.tile(tm.makeStringIndex(K).values, N)
+        index = MultiIndex.from_arrays([level1, level2])
+        self.s = Series(np.random.randn(N * K), index=index)
+        self.s_subset = self.s[::2]
+        self.s_subset_no_cache = self.s[::2].copy()
+
+        mi = MultiIndex.from_product([rng, range(100)])
+        self.s2 = Series(np.random.randn(len(mi)), index=mi)
+        self.s2_subset = self.s2[::2].copy()
+
+    def time_reindex_dates(self, shape):
+        execute(self.df.reindex(self.rng_subset))
+
+    def time_reindex_columns(self, shape):
+        execute(self.df2.reindex(columns=self.df.columns[1:5]))
+
+    def time_reindex_multiindex_with_cache(self, shape):
+        # MultiIndex._values gets cached
+        execute(self.s.reindex(self.s_subset.index))
+
+    def time_reindex_multiindex_no_cache(self, shape):
+        # Copy to avoid MultiIndex._values getting cached
+        execute(self.s.reindex(self.s_subset_no_cache.index.copy()))
+
+    def time_reindex_multiindex_no_cache_dates(self, shape):
+        # Copy to avoid MultiIndex._values getting cached
+        execute(self.s2_subset.reindex(self.s2.index.copy()))
+
+
+class TimeReindexMethod:
+
+    params = [
+        get_benchmark_shapes("TimeReindexMethod"),
+        ["pad", "backfill"],
+        [IMPL.date_range, IMPL.period_range],
+    ]
+    param_names = ["shape", "method", "constructor"]
+
+    def setup(self, shape, method, constructor):
+        N = shape[0]
+        self.idx = constructor("1/1/2000", periods=N, freq="1min")
+        self.ts = Series(np.random.randn(N), index=self.idx)[::2]
+
+    def time_reindex_method(self, shape, method, constructor):
+        execute(self.ts.reindex(self.idx, method=method))
+
+
+class TimeFillnaMethodSeries:
+
+    params = [get_benchmark_shapes("TimeReindexMethod"), ["pad", "backfill"]]
+    param_names = ["shape", "method"]
+
+    def setup(self, shape, method):
+        N = shape[0]
+        self.idx = IMPL.date_range("1/1/2000", periods=N, freq="1min")
+        ts = Series(np.random.randn(N), index=self.idx)[::2]
+        self.ts_reindexed = ts.reindex(self.idx)
+        self.ts_float32 = self.ts_reindexed.astype("float32")
```

Comment:
So that previous operations do not affect the measured time.
```suggestion
        self.ts_reindexed = ts.reindex(self.idx)
        self.ts_float32 = self.ts_reindexed.astype("float32")
        execute(self.ts_reindexed)
        execute(self.ts_float32)
```

Reply:
Added execute to setup of all newly added classes